.github/workflows/codeql-analysis.yml

```diff
@@ -36,6 +36,11 @@ jobs:
           # a pull request then we can check out the head.
           fetch-depth: 2
 
+      - name: Install Go from go.mod
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+
```

Comment:
Do we need this here?
Previously we didn't install go in this workflow

Reply:
I added a comment with link to the issue regarding this question.

For everyone reading, we need it because gh actions runner uses go1.20 by default, so this CodeQL action will not pass when the go version in go.mod file is higher than that. That's why we need to install go from go.mod by our own.